plugin/src/main/java/org/wso2/carbon/uuf/maven/ComponentMojo.java

```diff
@@ -24,43 +24,87 @@
+import org.wso2.carbon.uuf.maven.util.ZipCreator;
 
-import java.nio.file.Paths;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 /**
- * UUF Component creation Mojo that generates the osgi-imports when this component uses any OSGi imports
- * and creates the component zip archive for the given component project.
+ * UUF Component creation Mojo that generates the osgi-imports when this component uses any OSGi imports and creates the
+ * component zip archive for the given component project.
  *
  * @since 1.0.0
  */
 @Mojo(name = "create-component", inheritByDefault = false, requiresDependencyResolution = ResolutionScope.COMPILE,
       threadSafe = true, defaultPhase = LifecyclePhase.PACKAGE)
-public class ComponentMojo extends AbstractZipMojo {
+public class ComponentMojo extends AbstractUUFMojo {
+
+    protected static final String CONFIGURATION_IMPORT_PACKAGE = "Import-Package";
+    protected static final String FILE_CONFIG = "config.yaml";
+    protected static final String FILE_COMPONENT_MANIFEST = "component.yaml";
+
+    /**
+     * The temporary directory for UUF Maven plugin.
+     */
+    @Parameter(defaultValue = "${project.build.directory}/uuf-temp/", readonly = true, required = true)
+    protected String tempDirectoryPath;
 
     /**
      * OSGi {@code <Import-Package>} instructions for this UUF Component.
      */
     @Parameter(readonly = true, required = false)
-    private Map<String, String> instructions;
+    protected Map<String, String> instructions;
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public String getZipBaseDirectory() {
-        int lastIndex = artifactId.lastIndexOf(".");
-        if (lastIndex > -1) {
-            return artifactId.substring(lastIndex + 1);
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        // Validation: Packaging type should be 'uuf-component'.
+        if (!ARTIFACT_TYPE_UUF_COMPONENT.equals(packaging)) {
+            throw new MojoExecutionException(
+                    "Packaging type of an UUF Component should be '" + ARTIFACT_TYPE_UUF_COMPONENT +
+                            "'. Instead found '" + packaging + "'.");
+        }
+        // Validation: Parse configuration file to make sure it is a valid YAML file.
+        String configFilePath = pathOf(sourceDirectoryPath, FILE_CONFIG);
+        try {
+            new ConfigurationParser().parse(configFilePath);
+        } catch (ParsingException e) {
+            throw new MojoExecutionException(
+                    "Configuration file '" + configFilePath + "' of this UUF Component is invalid.", e);
+        }
+        // Validation: Parse component manifest file to make sure it is valid.
+        String componentManifestFilePath = pathOf(sourceDirectoryPath, FILE_COMPONENT_MANIFEST);
+        try {
+            new ComponentManifestParser().parse(componentManifestFilePath);
```

Comment:
How frequent is this operation? At a glance this looks like we are creating a new object and discarding it for the GC to collect later.